odoo/addons/base/ir/ir_model.py

```diff
@@ -281,6 +282,27 @@ def _check_selection(self, selection):
             raise UserError(_("The Selection Options expression is not a valid Pythonic expression."
                               "Please provide an expression in the [('key','Label'), ...] format."))
 
+    @api.constrains('name')
```

Comment:
Yes it does. The justification is that you can change the `name` of a field, but you cannot change its `state`. However being more explicit will not hurt in this case.